Packages/com.unity.inputsystem/InputSystem/Plugins/XR/TrackedPoseDriver.cs

```diff
@@ -287,21 +445,100 @@ protected void OnDisable()
         protected virtual void OnDestroy()
         {
 #if UNITY_INPUT_SYSTEM_ENABLE_VR && ENABLE_VR
-            if (HasStereoCamera())
+            if (HasStereoCamera(out var cameraComponent))
             {
-                UnityEngine.XR.XRDevice.DisableAutoXRCameraTracking(GetComponent<Camera>(), false);
+                UnityEngine.XR.XRDevice.DisableAutoXRCameraTracking(cameraComponent, false);
             }
 #endif
         }
 
+        /// <summary>
+        /// The callback method called after the Input System has completed an update and processed all pending events.
+        /// </summary>
+        /// <seealso cref="InputSystem.onAfterUpdate"/>
         protected void UpdateCallback()
         {
+            if (m_IsFirstUpdate)
+            {
+                // Update current input values if this is the first update since becoming enabled
+                // since the performed callbacks may not have been executed
+                if (m_PositionInput.action != null)
+                    m_CurrentPosition = m_PositionInput.action.ReadValue<Vector3>();
+
+                if (m_RotationInput.action != null)
+                    m_CurrentRotation = m_RotationInput.action.ReadValue<Quaternion>();
+
+                ReadTrackingState();
+
+                m_IsFirstUpdate = false;
+            }
+
             if (InputState.currentUpdateType == InputUpdateType.BeforeRender)
                 OnBeforeRender();
             else
                 OnUpdate();
         }
 
+        void ReadTrackingState()
+        {
+            var trackingStateAction = m_TrackingStateInput.action;
+            if (trackingStateAction != null && !trackingStateAction.enabled)
+            {
+                // Treat a disabled action as the default None value for the ReadValue call
+                m_CurrentTrackingState = TrackingStates.None;
+                return;
+            }
+
+            if (trackingStateAction == null || trackingStateAction.m_BindingsCount == 0)
+            {
+                // Treat an Input Action Reference with no reference the same as
+                // an enabled Input Action with no authored bindings, and allow driving the Transform pose.
+                m_CurrentTrackingState = TrackingStates.Position | TrackingStates.Rotation;
+                return;
+            }
+
+            // Grab state.
+            var actionMap = trackingStateAction.GetOrCreateActionMap();
```

Comment:
If we cannot assume m_TrackingStateInput is properly set, we need some extra null checks here unfortunately...
```suggestion
            if (m_TrackingStateInput != null && m_TrackingStateInput.action != null && !trackingStateAction.enabled)
            {
                // Treat a disabled action as the default None value for the ReadValue call
                m_CurrentTrackingState = TrackingStates.None;
                return;
            }
           
            if (m_TrackingStateInput == null || m_TrackingStateInput.action == null || trackingStateAction.m_BindingsCount == 0)
            {
                // Treat an Input Action Reference with no reference the same as
                // an enabled Input Action with no authored bindings, and allow driving the Transform pose.
                m_CurrentTrackingState = TrackingStates.Position | TrackingStates.Rotation;
                return;
            }

            // Grab state.
            var trackingStateAction = m_TrackingStateInput.action;
            var actionMap = trackingStateAction.GetOrCreateActionMap();
```